snippets/csharp/api/system.security.cryptography/hashalgorithm/example1.cs

```diff
@@ -0,0 +1,66 @@
+using System;
+using System.Security.Cryptography;
+using System.Text;
+
+class Program
+{
+    static void Main()
+    {
+        string source = "Hello World!";
+        using (SHA256 sha256Hash = SHA256Managed.Create())
+        {
+            string hash = GetSHA256Hash(sha256Hash, source);
+
+            Console.WriteLine($"The SHA256 hash of {source} is: {hash}.");
+
+            Console.WriteLine("Verifying the hash...");
+
+            if (VerifySHA256Hash(sha256Hash, source, hash))
+            {
+                Console.WriteLine("The hashes are the same.");
+            }
+            else
+            {
+                Console.WriteLine("The hashes are not same.");
+            }
+        }
+    }
+
+    static string GetSHA256Hash(SHA256 sha256Hash, string input)
```

Comment:
I'd expect `GetSHA256Hash` to own creating the hash object.  `GetHash(HashAlgorithm hashAlgorithm, string input)`, on the other hand, makes sense.